docs/data/data-grid/row-spanning/RowSpanning.tsx

```diff
@@ -0,0 +1,138 @@
+import * as React from 'react';
+import Box from '@mui/material/Box';
+import { DataGrid, GridColDef } from '@mui/x-data-grid';
+import FormControlLabel from '@mui/material/FormControlLabel';
+import Switch from '@mui/material/Switch';
+
+export default function RowSpanning() {
+  const [enabled, setEnabled] = React.useState(true);
+
+  return (
+    <Box sx={{ width: '100%' }}>
+      <FormControlLabel
+        checked={enabled}
+        onChange={(event) => setEnabled((event.target as HTMLInputElement).checked)}
+        control={<Switch />}
+        label="Enable row spanning"
+      />
+      <Box sx={{ height: 300 }}>
+        <DataGrid
+          rows={rows}
+          columns={columns}
+          initialState={{
+            pagination: {
+              paginationModel: {
+                pageSize: 10,
+              },
+            },
+          }}
+          density="compact"
+          showCellVerticalBorder
```

Comment:
```suggestion
          showCellVerticalBorder
          showColumnVerticalBorder
```

Nit: I think these would look slightly nicer with both border props enabled